extension/src/content_scripts/google_classroom/FacialDetection.js

```diff
@@ -27,32 +27,64 @@ async function renderPrediction() {
                     predictions[i].landmarks = predictions[i].landmarks.arraySync();
                 }
             }
-  
-            const start = predictions[i].topLeft;
-            const end = predictions[i].bottomRight;
-            const size = [end[0] - start[0], end[1] - start[1]];
-            GVars.ctx.strokeStyle = "white";
-            GVars.ctx.strokeRect(start[0], start[1], size[0], size[1]);
-  
-            if (annotateBoxes) {
-                const landmarks = predictions[i].landmarks;
-                GVars.ctx.fillStyle = "white";
-                for (let j = 0; j < landmarks.length; j++) {
-                    const x = landmarks[j][0];
-                    const y = landmarks[j][1];
-                    GVars.ctx.fillRect(x, y, 5, 5);
-                }
+
+            // Convert Image to tensor
+            const faceInput = tf.browser.fromPixels(GVars.video, 3)
+                .expandDims(0);   
+            const input = tf.tensor4d(Array.from(faceInput.dataSync()), [1, faceInput.shape[1], faceInput.shape[2], 3]);
+
+            // crop face based on facial detection results
+            const normalizeStart = tf.div(predictions[i].topLeft, input.shape.slice([1], [2]));
+            const normalizeEnd = tf.div(predictions[i].bottomRight, input.shape.slice([1], [2]));
+            const boxes = tf.concat([normalizeStart, normalizeEnd]).reshape([-1, 4]);
+            const cropFace = tf.image.cropAndResize(faceInput, boxes, [0], [150, 150]);
+
+            // preprocessing grayscale
+            const r = cropFace.slice([0, 0, 0, 0], [1, cropFace.shape[1], cropFace.shape[2], 1]);
+            const b = cropFace.slice([0, 0, 0, 1], [1, cropFace.shape[1], cropFace.shape[2], 1]);
+            const g = cropFace.slice([0, 0, 0, 2], [1, cropFace.shape[1], cropFace.shape[2], 1]);
+            const avg = tf.div(tf.addN([r, b, g]), tf.tensor([3]));
+            const grayInput = tf.concat([avg, avg, avg], 3);
+            
+            // expression prediction
+            let prediction = await GVars.facialExpressionModel.predict(grayInput);
+            console.log(prediction.dataSync()[0]);
+            let confidence = Math.abs(prediction.dataSync()[0] - .5) / .5;
+            let confidenceLabel = confidence < .5 ? true : false;
+            let label = prediction.dataSync()[0] < .5 ? true : false;
+            console.log(label);
```

Comment:
Since you inherit this code, you gotta make some additional changes.

use `debug` and make this more verbose, please do not just print a random bool in the console.